argoverse/evaluation/detection/utils.py

```diff
@@ -323,20 +320,28 @@ def filter_instances(
     return filtered_instances
 
 
-def rank(dts: List[ObjectLabelRecord]) -> Tuple[np.ndarray, np.ndarray]:
-    """Get the rankings for the detections, according to detector confidence.
+def rank(dts: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+    """Rank the detections in descending order according to score (detector confidence).
+
 
     Args:
-        dts: Detections (N,).
+        dts: Array of `ObjectLabelRecord` objects. (N,).
 
     Returns:
-        ranks: Ranking for the detections (N,).
-        scores: Detection scores (N,).
+        ranked_dts: Array of `ObjectLabelRecord` objects ranked by score (N,) where N <= MAX_NUM_BOXES.
+        ranked_scores: Array of floats sorted in descending order (N,) where N <= MAX_NUM_BOXES.
     """
-    scores = np.array([dt.score for dt in dts])
+    scores = np.array([dt.score for dt in dts.tolist()])
     ranks = scores.argsort()[::-1]
-    ranked_detections = dts[ranks]
-    return ranked_detections, scores[:, np.newaxis]
+
+    ranked_dts = dts[ranks]
+    ranked_scores = scores[ranks]
+
+    # Ensure the number of boxes considered per class is at most `MAX_NUM_BOXES`.
+    if ranked_dts.shape[0] > MAX_NUM_BOXES:
```

Comment:
[nit] We can avoid the if condition here.
Just `return ranked_dts[:MAX_NUM_BOXES], ranked_scores[:MAX_NUM_BOXES]`

Reply:
good point

Reply:
I think the former is a tiny bit faster from benchmarking, but they're both on the order of `1e-7 - 1e-6`, so I personally think removing the conditional would look better.